src/injector/middleware.go

```diff
@@ -0,0 +1,23 @@
+package injector
+
+import (
+	"github.com/inloco/kafka-elasticsearch-injector/src/metrics"
+	"github.com/inloco/kafka-elasticsearch-injector/src/models"
+	"time"
+)
+
+type instrumentingMiddleware struct {
```

Comment:
Why "instrumenting"? Why not "latencyPublisherMiddleware"? I'd go for a more descriptive name.

Reply:
I guess that's a better name, indeed. In case we need other metrics, creating a new middleware should be the way to go, instead of adding code to this one.